src/services/actions/apiService.js

```diff
@@ -0,0 +1,34 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react"
+
+const { VITE_API_KEY, VITE_API_URL } = import.meta.env;
+
+export const articleApi = createApi({
+    reducerPath: "articleApi",
+    baseQuery: fetchBaseQuery({
+        baseUrl: VITE_API_URL,
+        headers: {
+            "X-Api-Key": VITE_API_KEY,
+        },
```

Comment:
using the proxy server this will not work for long, try to pass the API key in the url

Reply:
Its a good idea, but the api key the url is not safe it can be easily seen, and I have to keep it private, also if I pass it in each url it will be a repedition. but why you say it will not work for long?